distro/src/avail/Avail.avail/Foundation.avail/Tuples.avail

```diff
@@ -2517,22 +2537,103 @@ Semantic restriction "split_before_" is
 |
 	Require: predicate accepts tupleType (check indices);
 	returnType ::= predicate's return type;
-	if ⎡tupleType⎤ = 0 then
+	if ⌈tupleType⌉ = 0 then
 	[
 		<<>, <>>'s type
 	]
 	else
 	[
 		sizes ::= ||tupleType||;
-		upper ::= [0..⎡sizes⎤ + 1);
+		upper ::= [0..⌈sizes⌉ + 1);
 		firstType ::=
 			<tupleType's leading types, tupleType's default type…|upper>;
 		secondType ::=
-			tuple type of tupleType[[1..⎡sizes⎤ + 1)..upper];
+			tuple type of tupleType[[1..⌈sizes⌉ + 1)..upper];
 		<firstType, secondType…|2>
 	]
 ];
 
+/**
+* Break a tuple into a tuple of tuples at elements matching a provided
+* delimiter.
+*
+* @method "explode_at element_«skipping empties»?"
+* @param "aTuple" "tuple"
+* @param "delimiter" "any"
+* @param "skipEmpties" "boolean"
+*        Whether to include empty subtuples in the result, as caused by
+*        adjacent occurrences of the delimiter, or a delimiter in initial or
+*        final position.
+* @returns "tuple*"
+*          A tuple of the subtuples of the original, as broken up by the
+*          matching delimiter, with that delimiting element removed.
+* @category "Tuples" "Transformers"
+*/
+Public stable method "explode_at element_«skipping empties»?" is
```

Comment:
In a vacuum I like "exploded" more but since I built it as extension of `split_before_`/`split_after_` I think that biased me. I'll ruminate.

I see an argument for pure transformations of immutable values preferring past-participle postfix names and more stateful, side-effect-laden methods preferring present imperative prefix names, but it probably still comes down to taste most of the time. E.g. `_split before_` doesn't sound as good to me (the present/past/past participle ambiguity of split is at least partially to blame).